// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`otel formatter errors encodes ClientReportedError Error as parseable JSON 1`] = `
Object {
  "attributes": Object {
    "agent": "Darwin x64",
    "device.agent": "Browser/5.0 (compatible; WXYZ 100.0; Doors LQ 81.4; Boat/1.0)",
    "error.message": "something broke",
    "error.name": "ClientReportedError",
    "error.stacktrace": "Error: something broke
    at methodA (resource-a.js:1:1)
    at methodB (resource-b.js:1:1)
",
    "request.address.uri": "https://example.com/page-the/error/occurred-on",
    "request.metaData.code": "500",
    "request.metaData.collectionMethod": "applicationError",
    "request.metaData.correlationId": "123abc",
    "request.metaData.moduleID": "healthy-frank",
  },
  "body": "something broke",
  "severityNumber": 17,
  "severityText": "ERROR",
}
`;

exports[`otel formatter errors encodes Server Reported Error as parseable JSON with nested metadata objects 1`] = `
Object {
  "attributes": Object {
    "agent": "Darwin x64",
    "error.message": "something broke",
    "error.name": "Error",
    "error.stacktrace": "Error: something broke
    at methodA (resource-a.js:1:1)
    at methodB (resource-b.js:1:1)
",
    "metaData.moduleID": "healthy-frank",
    "metaData.nestedObject.level1.level2.level3.level4": "logs nested objects correctly",
  },
  "body": "something broke",
  "severityNumber": 17,
  "severityText": "ERROR",
}
`;

exports[`otel formatter errors puts a lone error in the error field 1`] = `
Object {
  "agent": "Darwin x64",
  "error.message": "test error",
  "error.name": "Error",
  "error.stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
}
`;

exports[`otel formatter errors puts an error paired with a message in the error field 1`] = `
Object {
  "agent": "Darwin x64",
  "error.message": "test error",
  "error.name": "Error",
  "error.stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
}
`;

exports[`otel formatter errors puts an error with other arguments in the error field and uses only the other args for the body 1`] = `
Object {
  "agent": "Darwin x64",
  "error.message": "test error",
  "error.name": "Error",
  "error.stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
}
`;

exports[`otel formatter errors uses the error in the message when the error is paired with a message and other arguments 1`] = `
Object {
  "agent": "Darwin x64",
  "error.message": "test error",
  "error.name": "Error",
  "error.stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
}
`;

exports[`otel formatter errors uses the error in the message when the error is paired with a message and other arguments 2`] = `
"unable to do the thing Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7) 12 times"
`;

exports[`otel formatter types encodes request as parseable JSON 1`] = `
Object {
  "attributes": Object {
    "agent": "Darwin x64",
    "request.address.url": "https://example.org/server",
    "request.direction": "in",
    "request.metaData.correlationId": "123",
    "request.metaData.host": "example.org",
    "request.metaData.location": undefined,
    "request.metaData.method": "GET",
    "request.metaData.referrer": "https://example.org/other-page",
    "request.metaData.userAgent": "Browser/8.0 (compatible; WXYZ 100.0; Doors LQ 81.4; Boat/1.0)",
    "request.protocol": "https",
    "request.statusCode": 200,
    "request.statusText": "OK",
    "request.timings.duration": 10,
    "request.timings.ttfb": 20,
  },
  "severityNumber": 17,
  "severityText": "ERROR",
}
`;

exports[`otel formatter types uses the default util.format when given an unknown type 1`] = `
Object {
  "attributes": Object {
    "agent": "Darwin x64",
  },
  "body": "{ type: 'yolo' }",
  "severityNumber": 17,
  "severityText": "ERROR",
}
`;
